src/cudamatrix/cu-vector.cc

```diff
@@ -711,6 +711,47 @@ void CuVectorBase<Real>::MulElements(const CuVectorBase<Real> &v) {
 
 template<>
 template<>
+void CuVectorBase<double>::CopyFromVec(const CuVectorBase<double> &src) {
+  KALDI_ASSERT(src.Dim() == dim_);
+#if HAVE_CUDA == 1
+  if (CuDevice::Instantiate().Enabled()) {
+    if (dim_ == 0) return;
+    Timer tim;
+    int dimBlock(CU2DBLOCK);
+    int dimGrid(n_blocks(dim_, CU2DBLOCK));
+    cuda_copy_from_vec(dimGrid, dimBlock, data_, src.data_, dim_);
+    CU_SAFE_CALL(cudaGetLastError());
+    CuDevice::Instantiate().AccuProfile(__func__, tim.Elapsed());
```

Comment:
`__func__` will evaluate to "CopyFromVec" for all four of these functions. [See here](https://gcc.gnu.org/onlinedocs/gcc/Function-Names.html). Obviously, the time each of the four functions takes will be different (since double datatypes have twice as many bits as float types). Someone else will have to comment on whether this is okay anyway.